Camera app: Add `--video-device` arg instead of hardcoding `/dev/video0`

samadDotDev opened the pull request:
#### Summary

This adds `AppOptions` for the Camera app for configurability of the app at startup, and first configuration this PR adds is the arg for `--video-device` (defaults to `/dev/video0`) so we don't hardcode this in a bunch of places.

#### Testing

`./out/linux-x64-camera-clang/chip-camera-app ... --video-device /dev/video0`

```
[1750104633.856] [432754:432754] [CAM] Matter Camera Linux App: ApplicationInit() with Video Device Path: /dev/video0

[1750104679.945] [432754:432754] [CAM] Allocated snapshot stream with ID: 1
[1750104679.945] [432754:432754] [CAM] Detected USB camera: /dev/video0
```


`./out/linux-x64-camera-clang/chip-camera-app ... --video-device /dev/video1` 
(no stream allocation made in the following case)
```
[1750109690.322] [542006:542006] [CAM] Matter Camera Linux App: ApplicationInit() with Video Device Path: /dev/video1
```

`./out/linux-x64-camera-clang/chip-camera-app ... --video-device /dev/video2`
(No device at this FD/path)
```
[1750109801.542] [542830:542830] [CAM] Matter Camera Linux App: ApplicationInit() with Video Device Path: /dev/video2
[1750109801.544] [542830:542830] [CAM] Error opening video device: No such file or directory at /dev/video2
```

samadDotDev committed:
Add app option to get video device path

samadDotDev committed:
Propagate the video device path to camera app

samadDotDev committed:
Apply gemini suggestions for error return & string reuse

samadDotDev commented on examples/camera-app/linux/src/camera-device.cpp:
```diff
@@ -340,6 +333,14 @@ CameraDevice::~CameraDevice()
+    InitializeCameraDevice();
+    InitializeStreams();
+    mWebRTCProviderManager.Init();
+    return CHIP_NO_ERROR;
```
Removed the error type from `Init()` signature to keep the existing behavior. This is required at the moment because otherwise the camera-app in CI would exit when there is no video device to initialize, so keeping the current behavior of logging error on that as opposed to crashing the app in https://github.com/project-chip/connectedhomeip/pull/39552/commits/b9c042d7f101c607ccc4ec92f19e55d3e853a24f

samadDotDev commented on examples/camera-app/linux/main.cpp:
```diff
@@ -29,7 +31,10 @@ CameraDevice gCameraDevice;
-    ChipLogProgress(Camera, "Matter Camera Linux App: ApplicationInit()");
+    ChipLogProgress(Camera, "Matter Camera Linux App: ApplicationInit() with Video Device Path: %s",
+                    AppOptions::GetVideoDevicePath().c_str());
+    gCameraDevice.SetVideoDevicePath(AppOptions::GetVideoDevicePath());
```
Addressed in https://github.com/project-chip/connectedhomeip/pull/39552/commits/b9c042d7f101c607ccc4ec92f19e55d3e853a24f

yufengwangca commented on examples/camera-app/linux/include/camera-device.h:
```diff
@@ -66,6 +66,8 @@ static constexpr uint16_t kMaxImageRotation          = 359; // Spec constraint
 
 namespace Camera {
 
+inline constexpr const char * kDefaultVideoDevicePath = "/dev/video0";
```
We can drop inline here since kDefaultVideoDevicePath is only used in one compiling unit.

If you need kDefaultVideoDevicePath shared across all translation units (for example, you take the variable’s address and compare it), add inline, but we are not in that case.

yufengwangca commented:
We don't need to implement AppOptions.h/AppOptions.cpp by ourself, add the new option to ./platform/linux/Options.cpp under flag ENABLE_CAMERA_SERVER, we already have one config param '--camera-deferred-offer'

#if ENABLE_CAMERA_SERVER
    kDeviceOption_Camera_DeferredOffer,
#endif

samadDotDev committed:
Remove `AppOptions`, add arg to linux options

samadDotDev commented:
Whoops, didn't know about this being used with Camera gating in common linux options, thanks @yufengwangca for pointing this out! I reverted the `AppOptions` changes and introduced the arg in linux options as part of https://github.com/project-chip/connectedhomeip/pull/39552/commits/b72f0a71d0ed06c05f320b25ffd18449c020b687


Updated help options:

```
❯ ./out/linux-x64-camera-clang/chip-camera-app --help

GENERAL OPTIONS

...

  --camera-deferred-offer
       Indicates the delayed processing hint of the WebRTC Provider.

  --camera-video-device <path>
       Path to a V4L2 video capture device (default: /dev/video0).

```

Updated logs:

```
❯ ./out/linux-x64-camera-clang/chip-camera-app ...

[1750182622.806] [1213779:1213779] [CAM] Matter Camera Linux App: ApplicationInit()
[1750182622.806] [1213779:1213779] [CAM] Using default video device path: /dev/video0
```

```
❯ ./out/linux-x64-camera-clang/chip-camera-app ... --camera-video-device /dev/video1

[1750177603.553] [1162716:1162716] [CAM] Matter Camera Linux App: ApplicationInit()
[1750177603.553] [1162716:1162716] [CAM] Using video device path from options: /dev/video1
```

samadDotDev committed:
Merge branch 'master' into fix/camera-path-arg
